### PR TITLE
Ament lint documentation

### PR DESCRIPTION
**Pull request by fjanguita**
· Documentation for configuring, running and passing ament_lint_tests added.
 · Documentation structure at 'Develop Guide' refactored to include new section.
 · Documentation on new section for setting up VSCode for testing added.

### CODE REVIEW

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
+.. _development_guide_new_pkg:
+
+------------------------
+Developing a New Package
+------------------------
+
+TDB
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:7]: Remove this

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+
+
+.. _development_guide_new_pkg_architecture:
+
+Architecture of a New Package
+=============================
+
+Every new package should have the following architecture:
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:16]: Some changes:
- Add LICENSE and CONTRIBUTING.md
- Header files should be in /include/package_name. Also add a package_name.hpp header
- In src, a package_name.cpp and package_name_node.cpp  should appears
- Set README.md to uppercase

Separate for cpp, python and hybrid packages

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+Aerostack2 follows the `ROS 2 code style and language versions <https://docs.ros.org/en/humble/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html>`_.
+
+
+
+.. _development_guide_new_pkg_code_style_c++:
+
+C++
+---
+
+Code should follows the `google c++ style guide <https://google.github.io/styleguide/cppguide.html>`_.
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:52]: We use ROS 2 code style, but while ROS 2 says it uses Google code style, they use snake_case for functions, while Google use camelCase. We use camelCase

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+Every file should start with the Licence text, which looks like this:
+
+.. code-block:: c++
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:56]: This is not correct, is a deprecated copyright.

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+Code Functional Test
+--------------------
+
+In aerostack2 we use googletest (GTest) library to perform unit tests across the packages.
+GTest complete documentation about how to write your own unit tests can be found at:
+https://github.com/google/googletest
+
+In order to compile this tests some lines must be added into a **NEW** ``CMakeLists.txt`` file located in a ``tests/`` folder.
+
+.. code-block:: cmake
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:255]: Deprecated also. See: https://github.com/aerostack2/aerostack2/blob/main/as2_aerial_platforms/as2_platform_multirotor_simulator/tests/CMakeLists.txt

File: docs/_09_development/_develop_guide/_develop_guide/_setting_code.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    ],
+
+    // python
+    "[python]": {
+        "editor.defaultFormatter": "ms-python.autopep8",
+        "editor.formatOnSave": true,
+    },
+    "autopep8.args": ["--max-line-length=99", "--ignore=''"],
+
+    // ROS2 uses as a linter flake8==4.0.1 (apt installation). Pip installation should be
```
Comment (pariaspe) [inline docs/_09_development/_develop_guide/_develop_guide/_setting_code.rst:69]: Afaik, this issue happens only with ROS 2 Humble. Newer versions of ROS 2 include upgraded flake8 (in ament_flake8). Thus, flake8 might be used from code too with the following settings:
```
    // flake8
    "flake8.args": ["--extend-ignore=B902,C816,D100,D101,D102,D103,D104,D105,D106,D107,D203,D212,D404,I202",
                    "--import-order-style=google", "--max-line-length=99", "--show-source=true",
                    "--statistics=true"],
```

**Review by pariaspe — CHANGES_REQUESTED** [review]

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+.. _development_guide_new_pkg_test_functional:
+
+Code Functional Test
+--------------------
+
+In aerostack2 we use googletest (GTest) library to perform unit tests across the packages.
+GTest complete documentation about how to write your own unit tests can be found at:
+https://github.com/google/googletest
+
+In order to compile this tests some lines must be added into a **NEW** ``CMakeLists.txt`` file located in a ``tests/`` folder.
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:253]: gtest must be added to package.xml also

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+.. code-block:: bash
+
+    $ colcon test 
+
+.. _development_guide_new_pkg_test_coverage:
+
+Code Coverage Test
+------------------
+
+TDB
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:320]: We use github workflow to do so. Remove this section

**Review by RPS98 — CHANGES_REQUESTED** [review]

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst (hunk 3 above)
Comment (fjanguita) [reply docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:52]: Is this the only difference in code style or should I mention that there are a few more?

Comment (RPS98) [reply docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:52]: ROS 2 code style says: "Google style guide says CamelCase, but the C++ std library’s style of snake_case is also allowed"
We use Google style (that affects mainly to functions and methods)

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+
+Every new C++ package should have the following architecture:
+
+.. code-block::
+
+    /package_name
+        /include
+            /package_name
+                ### header files ###
+                node_name(.hpp)
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:23]: Something like:
    /package_name
        /include
            /package_name
                header_files(.hpp)
                package_name.hpp
...
/src
   implementation_files(.cpp)
   package_name.cpp
   package_name_node.cpp
...
/tests
   package_name_gtest.cpp
   package_name_test.cpp

Note that gtest must end with _gtest.cpp and normal test executables with _test.cpp (see tests/CMakeLists.txt)

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst [hunk 10]
```diff
@@ -0,0 +1,378 @@
...
+        .gitignore
+
+.. _development_guide_new_pkg_code_style:
+
+Code Style
+==========
+
+Aerostack2 mainly follows the `ROS 2 code style and language versions <https://docs.ros.org/en/humble/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html>`_. 
+There are, though, some exceptions, as ROS 2 claims to follow Goggle code style but uses snake_case
+instead of camelCase. Aerostack2 uses camelCase.
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:105]: Aerostack2 uses camelCase. -> Aerostack2 uses Google code style.

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst (hunk 4 above)
Comment (RPS98) [reply docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:117]: Add to files header something like:
```
/**
 * @file file_name.cpp
 *
 * File description
 *
 * @author Author name <author email>
 */
```

**Review by RPS98 — CHANGES_REQUESTED** [review]

File: docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst (hunk 9 above)
Comment (fjanguita) [reply docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:23]: is it compulsory for every package to have a header file with the same name as the package? same for implementation and main files. Wouldn´t it be 'node_name.hpp', for example?

Comment (RPS98) [reply docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst:23]: The standar way of aerostack2 it this. There are some exceptions, but mainly we try to use the package name for simple packages that have only one node

File: docs/_09_development/_develop_guide/index.rst [hunk 11]
```diff
@@ -6,234 +6,9 @@ Develop Guide
 
 TDB
```
Comment (RPS98) [inline docs/_09_development/_develop_guide/index.rst:7]: Remove TDB for a description of the section and the toctree

**Review by RPS98 — CHANGES_REQUESTED** [review]

**Review by pariaspe — APPROVED** [review]

**Review by RPS98 — APPROVED** [review]

**Pull request merged**